### PR TITLE
Various janitoring

### PR DESCRIPTION
**Pull request by akva2**
Quell static analyzer and additional clang warnings.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/core/transport/implicit/transport_source.c [hunk 1]
```diff
@@ -97,6 +96,7 @@ create_transport_source(int nsrc, int nphase)
             src->pressure   = NULL;  src->flux       = NULL;
             src->saturation = NULL;  src->surfvolume = NULL;
 
+            int status;
```
Comment (atgeirr) [inline opm/core/transport/implicit/transport_source.c]: This is not pedantically correct C89 I think even though gcc accepts it. I'll not let it prevent merging, we might want to  start compiling these files with a c++ compiler anyway.

File: opm/upscaling/RelPermUtils.cpp [hunk 2]
```diff
@@ -888,7 +888,7 @@ void RelPermUpscaleHelper::upscaleCapillaryPressure()
         to_double(options["saturationThreshold"]);
 
     auto largestSaturationInterval = Swor - Swir;
-    auto Ptestvalue                = Pcmax;
+    decltype(Pcmax) Ptestvalue;
```
Comment (atgeirr) [inline opm/upscaling/RelPermUtils.cpp]: This was a bit weird, can you explain why this is needed?

**Comment by atgeirr** [comment]
Thanks, will merge.

File: opm/upscaling/RelPermUtils.cpp (hunk 2 above)
Comment (akva2) [reply opm/upscaling/RelPermUtils.cpp]: Unused assignment. Apparently type deduction was wanted and i cant declare a bare auto.

**Pull request merged**